DARPA-ASKEM/TERArium
# feat: add facets to data explorer

### Summary
Add initial Facets integration allowing filtering of data explorer results using facet filters.

### Limitations
- CSS styling issues, e.g., the facet panel does not scroll
- Facet Vue components have lint disabled because of outdated code
- Input data for facets, e.g., aggregation of results, are all done on the client side on the fly because XDD for example does not provide native Facets support
- Refactoring of code my be needed to enable scalability with filtering and displaying dynamic facets for other types of resources such as simulations and workflows

### Testing
- open Data Explorer page
- Note the button at the left side representing the Facets panel collapsed by default when the ALL tab is selected
- Switch to the Models tab and note that the Facets panel has opened automatically and got populated with datacube/models related info
- Click on any of the Facets to filter the data and note how the information on the facets as well as the displayed list of results are updated.
- Click on the Articles tab and note how the Facets panels has adapted to the specific content of XDD Articles

Closes #29
